src/Compiler/CompilerOptions.cpp

```diff
@@ -590,16 +590,17 @@ bool parseCustomEnvFlagsCommandLineOption(
   // Use the default ONNX MLIR Environment variable, unless specified otherwise
   // by an argument, see below.
   std::string envVar = OnnxMlirEnvOptionName;
+  // VerboseOutput is not yet set, so scan ourselves.
+  bool verbose = false;
   // Customized version? -customEnvFlags=val and save its value.
-  for (int i = argc - 1; i > 1; --i) {
+  for (int i = argc - 1; i > 0; --i) {
```

Comment:
For command line arguments, when the same option is specified multiple times, typically the last one takes effect. That's probably why the arguments were scanned backwards originally. With the changes just made, you don't "break" at the first match (last instance) of `-customEnvFlags` or `--customEnvFlags`. Therefore, to keep the last-instance-takes-effect semantic, the arguments should now be scanned forwards instead of backwards.